base/random.jl

```diff
@@ -257,10 +257,21 @@ globalRNG() = GLOBAL_RNG
 Pick a random element or array of random elements from the set of values specified by `S`; `S` can be
 
 * an indexable collection (for example `1:n` or `['x','y','z']`), or
+* a `Dict`, a `Set` or an `IntSet`, or
 * a type: the set of values to pick from is then equivalent to `typemin(S):typemax(S)` for
   integers (this is not applicable to `BigInt`), and to ``[0, 1)`` for floating point numbers;
 
 `S` defaults to `Float64`.
+
+```jldoctest
+julia> srand(0); rand(Int, 2)
```

Comment:
Downside of that is that they wont be tested. On the other hand, these functions might be fundamental enough that they wont change. `julia-repl` is fine IMO.